include/zephyr/drivers/regulator.h

```diff
@@ -279,6 +287,38 @@ static inline int regulator_parent_dvs_state_set(const struct device *dev,
 	return api->dvs_state_set(dev, state);
 }
 
+/**
+ * @brief Set a power mode.
+ *
+ * Some PMICs feature low power modes which allow the system to save power.
+ * Exit from low power is normally by pin transition or timeout.
+ *
+ * This API can be used when the power state needs to be changed by software.
+ *
+ * @param dev Parent regulator device instance.
+ * @param state power state (vendor specific identifier).
+ * @param state_cfg Timeout configuration (vendor specific).
+ *
+ * @retval 0 If successful.
+ * @retval -ENOTSUP If given state is not supported.
+ * @retval -EPERM If state can't be changed by software.
+ * @retval -ENOSYS If function is not implemented.
+ * @retval -errno In case of any other error.
+ */
+static inline int regulator_parent_power_state_set(const struct device *dev,
```

Comment:
I'm a bit confused what the intention of this API is. This looks like a feature where the entire regulator can be moved from a high to low power state (via a pin transition or via software). The [regulator_parent_dvs_state_set](https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/drivers/regulator.h#L269) API was designed to cover the use case where an entire regulator needs to be reconfigured. Could this feature of the NPM1300 fit into that API?